### PR TITLE
feat(unlock-app) - add supplier details settings page

### PR DESCRIPTION
**Pull request by kalidiagne**
# Description

**Supplier details in settings page**

Screenshot 2023-01-25 at 18 54 30

**Supplier details settings page**
Screenshot 2023-01-23 at 16 03 56

**Supplier details settings page (non manager)**
Screenshot 2023-01-23 at 16 05 01

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-app/src/components/interface/locks/Settings/elements/SettingPayments.tsx [hunk 1]
```diff
@@ -61,6 +62,19 @@ export const SettingPayments = ({
           lock={lock}
         />
       </SettingCard>
+
+      <SettingCard
+        label="Receipts"
+        description="Save Supplier details for Receipts"
```
Comment (julien51) [inline unlock-app/src/components/interface/locks/Settings/elements/SettingPayments.tsx:68]:
```suggestion
        description="Update the supplier information to be shown on receipts."
```

**Review by julien51 — APPROVED** [review]
LGTM!

File: unlock-app/src/components/interface/locks/Settings/forms/ReceiptBaseForm.tsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    {
+      onSuccess: (res) => {
+        if (res.data) {
+          reset(res.data)
+        }
+      },
+      enabled: isManager,
+    }
+  )
+
```
Comment (searchableguy) [inline unlock-app/src/components/interface/locks/Settings/forms/ReceiptBaseForm.tsx:75]: Could extract these into hooks similar to what I did for metadata? Makes components much easier to read and reuse.

File: unlock-app/src/components/interface/locks/Settings/forms/ReceiptBaseForm.tsx [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        <Input disabled={disabledInput} label="City" {...register('city')} />
+        <Input disabled={disabledInput} label="State" {...register('state')} />
+        <Input disabled={disabledInput} label="Zip" {...register('zip')} />
+        <Input
+          disabled={disabledInput}
+          label="Country"
+          {...register('country')}
+        />
+      </div>
+
```
Comment (searchableguy) [inline unlock-app/src/components/interface/locks/Settings/forms/ReceiptBaseForm.tsx:118]: No error={errors.abc.message} passed in any? 

Should pass it even if we do not have any validation yet. We should add some here.

**Review by searchableguy — APPROVED** [review]
LGTM. A few comments.

File: unlock-app/src/components/interface/locks/Settings/forms/ReceiptBaseForm.tsx (hunk 3 above)
Comment (kalidiagne) [reply unlock-app/src/components/interface/locks/Settings/forms/ReceiptBaseForm.tsx:118]: All the fields are optional for now

File: unlock-app/src/components/interface/locks/Settings/forms/ReceiptBaseForm.tsx [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    >
+      <div className="grid grid-cols-2 gap-4">
+        <Input
+          disabled={disabledInput}
+          label="Supplier name"
+          {...register('supplierName')}
+        />
+        <Input disabled={disabledInput} label="VAT" {...register('vat')} />
+
+        <div className="col-span-2">
```
Comment (julien51) [inline unlock-app/src/components/interface/locks/Settings/forms/ReceiptBaseForm.tsx:92]: Let's move this box at the bottom of the form please

**Pull request merged**